lib/spack/spack/test/spec_semantics.py

```diff
@@ -393,70 +394,70 @@ def test_unsatisfiable_variant_types(self):
             constraint_spec="multivalue-variant ~foo",
             target_concrete=True,
         )
 
     def test_satisfies_unconstrained_variant(self):
         # only asked for mpich, no constraints.  Either will do.
-        check_satisfies("mpich+foo", "mpich")
-        check_satisfies("mpich~foo", "mpich")
-        check_satisfies("mpich foo=1", "mpich")
+        constrain_or_raise("mpich+foo", "mpich")
+        constrain_or_raise("mpich~foo", "mpich")
+        constrain_or_raise("mpich foo=1", "mpich")
 
     def test_unsatisfiable_variants(self):
         # This case is different depending on whether the specs are concrete.
 
         # 'mpich' is not concrete:
-        check_satisfies("mpich", "mpich+foo", False)
-        check_satisfies("mpich", "mpich~foo", False)
-        check_satisfies("mpich", "mpich foo=1", False)
-        check_satisfies("mpich", "mpich++foo", False)
-        check_satisfies("mpich", "mpich~~foo", False)
-        check_satisfies("mpich", "mpich foo==1", False)
+        constrain_or_raise("mpich", "mpich+foo", False)
+        constrain_or_raise("mpich", "mpich~foo", False)
+        constrain_or_raise("mpich", "mpich foo=1", False)
+        constrain_or_raise("mpich", "mpich++foo", False)
+        constrain_or_raise("mpich", "mpich~~foo", False)
+        constrain_or_raise("mpich", "mpich foo==1", False)
 
         # 'mpich' is concrete:
-        check_unsatisfiable("mpich", "mpich+foo", True)
-        check_unsatisfiable("mpich", "mpich~foo", True)
-        check_unsatisfiable("mpich", "mpich foo=1", True)
-        check_unsatisfiable("mpich", "mpich++foo", True)
-        check_unsatisfiable("mpich", "mpich~~foo", True)
-        check_unsatisfiable("mpich", "mpich foo==1", True)
+        check_cannot_constrain("mpich", "mpich+foo", True)
+        check_cannot_constrain("mpich", "mpich~foo", True)
+        check_cannot_constrain("mpich", "mpich foo=1", True)
+        check_cannot_constrain("mpich", "mpich++foo", True)
+        check_cannot_constrain("mpich", "mpich~~foo", True)
+        check_cannot_constrain("mpich", "mpich foo==1", True)
 
     def test_unsatisfiable_variant_mismatch(self):
         # No matchi in specs
-        check_unsatisfiable("mpich~foo", "mpich+foo")
-        check_unsatisfiable("mpich+foo", "mpich~foo")
-        check_unsatisfiable("mpich foo=True", "mpich foo=False")
-        check_unsatisfiable("mpich~~foo", "mpich++foo")
-        check_unsatisfiable("mpich++foo", "mpich~~foo")
-        check_unsatisfiable("mpich foo==True", "mpich foo==False")
+        check_cannot_constrain("mpich~foo", "mpich+foo")
+        check_cannot_constrain("mpich+foo", "mpich~foo")
+        check_cannot_constrain("mpich foo=True", "mpich foo=False")
+        check_cannot_constrain("mpich~~foo", "mpich++foo")
+        check_cannot_constrain("mpich++foo", "mpich~~foo")
+        check_cannot_constrain("mpich foo==True", "mpich foo==False")
 
     def test_satisfies_matching_compiler_flag(self):
```

Comment:
there's still a lot of left-over `satisfies` jargon in tests that use `intersect`

Reply:
Tests have been refactored, jargon shouldn't be there anymore